File: ios/RunningApp/Models/ApiResponse.swift
import Foundation

// MARK: - API统一响应格式
struct ApiResponse<T: Codable>: Codable {
    let code: Int
    let message: String
    let data: T?
}

// MARK: - 分页响应
struct PageResponse<T: Codable>: Codable {
    let list: [T]
    let total: Int
    let page: Int
    let pageSize: Int

    enum CodingKeys: String, CodingKey {
        case list, total, page
        case pageSize = "page_size"
    }
}

// MARK: - 空响应
struct EmptyResponse: Codable {}

File: android/app/src/main/java/com/runningapp/data/remote/model/ApiResponse.kt
package com.runningapp.data.remote.model

import com.google.gson.annotations.SerializedName

/**
 * API统一响应格式
 */
data class ApiResponse<T>(
    @SerializedName("code")
    val code: Int,
    @SerializedName("message")
    val message: String,
    @SerializedName("data")
    val data: T?
)

/**
 * 分页响应
 */
data class PageResponse<T>(
    @SerializedName("list")
    val list: List<T>,
    @SerializedName("total")
    val total: Int,
    @SerializedName("page")
    val page: Int,
    @SerializedName("page_size")
    val pageSize: Int
)

/**
 * 空响应
 */
class EmptyResponse
